server/datastore/mysql/software.go

```diff
@@ -3264,11 +3269,14 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 						// we want to treat the installed vpp app as a regular software title
 						delete(bySoftwareTitleID, s.ID)
 					}
+					byVPPAdamID[*s.VPPAppAdamID] = s
```

Comment:
Because of the way we track installed vpp software, we need this line. A vpp app present in `host_software` needs to be put into the `byVPPAdamID` slice so we look treat it like a vpp app vs a software_installer when we do the IN clauses later in this method to gather the apps to be returned.  The reason this is here and not under the `hostInstalledVppsApps` is because we don't filter by mdm enrolled in the `hostInstalledVppsApps` query. So when we are filtering for the self-service tab we would erroneously get vpp apps if the host was not mdm enrolled. https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software_test.go#L5372-L5377